File: generator/src/java/darkness/generator/scripts/uka17/WaterFill.java
package darkness.generator.scripts.uka17;

import darkness.generator.api.BulbGroup;
import darkness.generator.api.BulbRGB;

import darkness.generator.api.effects.WeightedStrobe;

import java.awt.*;

import java.util.ArrayList;
import java.util.SortedMap;
import java.util.TreeMap;
import java.util.Map.Entry;

public class WaterFill extends BaseScript {
    @Override
    public void run() {
        super.run();

        Color c = new Color(0, 127, 255);
        int t = 3;
        int fade_ext = 7;

        float [] positionBuffer;
	SortedMap<Float, ArrayList<BulbRGB>> orderedBulbs;
        for (BulbGroup letter : letters) {
		orderedBulbs = new TreeMap<Float, ArrayList<BulbRGB>>();
		ArrayList<BulbRGB> bulbList;
	        for (BulbRGB bulb : letter) {
		        positionBuffer = bulb.getPosition();
			bulbList = orderedBulbs.getOrDefault(positionBuffer[1], new ArrayList<BulbRGB>());
			bulbList.add(bulb);
			orderedBulbs.put(positionBuffer[1], bulbList);
	        }
		
		float lastHeight;
		for (Entry<Float, ArrayList<BulbRGB>> orderedBulbsElement : orderedBulbs.entrySet()) {
			for (BulbRGB bulb : orderedBulbsElement.getValue()) {
				rgbFade(bulb, c, t+fade_ext);
			}
			skip(t);
			lastHeight = orderedBulbsElement.getKey();
		}
		// TODO: filling drops through thingy missing
            skip(fade_ext);
	    }
	skip(10);
    }
}

File: generator/src/kotlin/darkness/generator/scripts/uka17/WaterFill.kt
package darkness.generator.scripts.uka17

import darkness.generator.api.BulbRGB
import java.awt.Color
import java.util.*

class WaterFill : BaseScript() {
    override suspend fun run() {
        super.run()

        val c = Color(0, 127, 255)
        val t = 3
        val fade_ext = 7

        var positionBuffer: FloatArray
        var orderedBulbs: SortedMap<Float, ArrayList<BulbRGB>>
        for (letter in letters) {
            orderedBulbs = TreeMap()
            var bulbList: ArrayList<BulbRGB>
            for (bulb in letter) {
                positionBuffer = bulb.position
                bulbList = orderedBulbs.getOrDefault(positionBuffer[1], ArrayList())
                bulbList.add(bulb)
                orderedBulbs[positionBuffer[1]] = bulbList
            }

            for (value in orderedBulbs.values) {
                for (bulb in value) {
                    rgbFade(bulb, c, t + fade_ext)
                }
                skip(t)
            }
            // TODO: filling drops through thingy missing
            skip(fade_ext)
        }
        skip(10)
    }
}
